pkg/apis/clusterregistry/validation/validation.go

```diff
@@ -32,5 +34,13 @@ func ValidateClusterUpdate(cluster, oldCluster *clusterregistry.Cluster) field.E
 		allErrs = append(allErrs, field.Invalid(field.NewPath("meta", "name"),
 			cluster.Name+" != "+oldCluster.Name, "cannot change cluster name"))
 	}
+	allErrs = append(allErrs, validateClusterName(cluster)...)
 	return allErrs
 }
+
+func validateClusterName(cluster *clusterregistry.Cluster) field.ErrorList {
+	if len(cluster.ClusterName) > 0 {
+		return field.ErrorList{field.Invalid(field.NewPath("meta", "clusterName"), "len(ClusterName) > 0", "clusterName is not used and must not be set.")}
```

Comment:
Isn't `field.NewPath("meta" ...)` actually `field.NewPath("metadata" ...)`? I see this is also repeated elsewhere in this file.

Reply:
Hmm. I was wondering that myself, but just assumed that it was correct. I just searched through the Kubernetes codebase, and found that this is usually `"metadata"`, though, so I've changed it here.

Reply:
Makes sense. Thanks for fixing this.